ipfx/epochs.py

```diff
@@ -0,0 +1,132 @@
+import numpy as np
+import time_series_utils as tsu
+
+
+# global constants
+#TODO: read them from the config file
+
+NOISE_EPOCH = 0.0015
+PRESTIM_STABILITY_EPOCH = 0.5
+POSTSTIM_STABILITY_EPOCH = 0.5
+LONG_RESPONSE_DURATION = 5  # this will count long ramps as completed
+
+
+def get_last_vm_epoch(idx1, hz):
+    """
+    Get epoch lasting LAST_STABILITY_EPOCH before the end of recording
+
+    Parameters
+    ----------
+    idx1
+    hz
+
+    Returns
+    -------
+
+    """
+    return idx1-int(POSTSTIM_STABILITY_EPOCH * hz), idx1
+
+
+def get_first_vm_noise_epoch(idx, hz):
+
+    return idx, idx + int(NOISE_EPOCH * hz)
+
+
+def get_last_vm_noise_epoch(idx1, hz):
+
+    return idx1-int(NOISE_EPOCH * hz), idx1
+
+
+def get_stability_vm_epoch(stim_start, hz):
+    num_steps = int(PRESTIM_STABILITY_EPOCH * hz)
+    if num_steps > stim_start-1:
+        num_steps = stim_start-1
+    elif num_steps <= 0:
+        return 0, 0
+    assert num_steps > 0, "Number of steps should be a positive integer"
+
+    return stim_start-1-num_steps, stim_start-1
+
+
+def get_recording_end_idx(v):
+
+    end_idx = np.nonzero(v)[0][-1]  # last non-zero index along the only dimension=0.
```

Comment:
If 'v' has only one dimension then you could just do `end_idx = np.flatnonzero(v)[-1]` and do away with the addition `[0]` indexing...

Same for line 58.